src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -4058,7 +4067,7 @@ protected override void PropertySet(PSProperty property, object setValue, bool c
         /// <returns>True if the property is settable.</returns>
         protected override bool PropertyIsSettable(PSProperty property)
         {
-            return !((PropertyCacheEntry)property.adapterData).readOnly;
+            return !((PropertyCacheEntry)property.adapterData).readOnly && !((PropertyCacheEntry)property.adapterData).initOnly;
```

Comment:
Why do we need new `initOnly` field if we do nothing specific for it? Maybe reuse `readOnly`?

Reply:
We don't strictly _need_ `initOnly` to make this change, we could make `readOnly` reflect it - but IMO it makes the adapter code easier to read and reason about than if we reuse `readOnly`.

Reply:
Looking on the `initOnly` I expect we do something specific for a scenario but we don't. So This is more misleading than beneficial.
Are there scenarios where the `initOnly` could be useful?

Reply:
One future scenario I had in mind was [surfacing it to the public API so tools like `Get-Member` can indicate it](https://github.com/PowerShell/PowerShell/issues/13819#issuecomment-949019584). That would require more feedback from the language WG and more rigorous testing (we'd be breaking public contract by introducing new concept in user-facing ETS facade, tools that parse metadata output might break, etc.), so that's beyond the scope of this PR.

Reply:
@IISResetMe Thanks for clarify. Since it is under question I don't think it's worth adding code that is not used.

Reply:
What do you mean by "not used"? I use it - to read and reason about what I've just implemented :)

Reply:
I mean, there is no way that `initOnly` would be used without `readOnly` now and there is no clear blueprint for the future. We will add this field when it is really needed.

Reply:
I just gave you [a clear blueprint for the future](https://github.com/PowerShell/PowerShell/pull/16520#discussion_r757668466), and explained that _I_ need this field now, to understand the code.

Reply:
If the concern is about adding another 4 bytes per allocated cache entry, then we can replace `readOnly`, `writeOnly` and `initOnly` with a single enum value instead...

Reply:
`bool` only takes up a byte in managed memory, so more than likely this just makes use of what was otherwise padding anyway.

I think it makes sense to keep if the plan is to surface it in `Get-Member` later. Since the field isn't a public API the risk is low. Also leaves room for a more specific message on attempted write.

Reply:
1 additional byte would move us from 52 -> 53 bytes of managed memory on x64 - but I guess the smallest cache line factor is 8 or 16 anyway :)